src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1817,13 +1826,24 @@ private function getNodeProcessor()
         $this->nodeProcessor = new NodeProcessor(
             $this->credentials->getUserID(),
             $this->getNamespacesObject(),
-            $this->getAutoLastModified()
+            $this->getAutoLastModified(),
+            $this->versionHandler
         );
 
         return $this->nodeProcessor;
     }
 
     /**
+     * Returns a handler for the versioning mechanism
+     *
+     * @return VersionHandler
+     */
+    private function getVersionHandler()
```

Comment:
You are right, just retained because it was a singleton in an earlier stage of development...